fpga/src/main/scala/spi/Spi.scala

```diff
@@ -0,0 +1,98 @@
+package spi
+
+import chisel3._
+
+import tools.helpers.log2
+
+class SpiSlave extends Module {
+  // This module accurately represents a SPI slave
+  // Mosi: Master Out, Slave In
+  // Miso: Master In, Slave Out
+  // SCLK: System Clock, 25MHz in our case
+  // CS: Chip Select
+  val io = IO(new Bundle {
+    val mosi = Input(Bool())
+    val miso = Output(Bool())
+    val sclk = Input(Bool())
+    val cs = Input(Bool())
+    val value = Output(UInt(dWidth.W))
+  })
+}
+
+/** [[Spi]]
+ * @param dWidth
+ */
+class Spi(dWidth: Int = 8) extends Module {
+
+  val io = IO(new Bundle {
+    // Spi signal
+    val spi = new SpiSlave
```

Comment:
I don't think this works??? If you want just an IO module, I think you should extend the `Bundle` and init IO with the extended class.